File: springboot-app/src/main/java/de/ngxa/restaurant/entity/Shop.java
package de.ngxa.restaurant.entity;

import de.ngxa.restaurant.constant.ShopType;
import lombok.Data;
import lombok.EqualsAndHashCode;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;

@Entity
@Data
@EqualsAndHashCode(callSuper = true)
@Table(name="NGXA_SHOP")
public class Shop extends BaseEntity {

	private String owner;
	private String name;

	private String alias;
	@Column(columnDefinition = "TEXT")
	private String longDesc;
	private String shortDesc;
	@Column(columnDefinition = "TEXT")
	private String openingTime;
	private String privacy;
	private String actualInfo;
	private Long minOrderValue = 0L;
	
	@OneToMany(cascade = CascadeType.ALL, fetch=FetchType.LAZY)
	@JoinColumn(name="shop_id", referencedColumnName="id")
	private List<ExtInfo> extInfo = new ArrayList<>();

	@OneToOne(cascade = CascadeType.ALL)
	private MetaInfo metaInfo;

	@OneToOne(cascade = CascadeType.ALL)
	private Contact contact;

	@OneToMany(cascade = CascadeType.ALL, fetch = FetchType.LAZY)
	@JoinColumn(name="shop_id", referencedColumnName="id")
	private List<OrderRegion> orderRegions = new ArrayList<>();

	@Column(nullable = false, columnDefinition = "boolean default true")
	private boolean isOrderAble = true;

	@Column(nullable = false, columnDefinition = "boolean default true")
	private boolean needIndex = true;

	@Enumerated(EnumType.STRING)
	@Column(columnDefinition="varchar")
	private ShopType shopType = ShopType.RESTAURANT;
	
	private String latitude;
	private String longitude;

}

File: manager-app/src/app/dto/shop.ts
import {BaseDto} from './base-dto';
import {ExtInfo} from './ext-info';
import {MetaInfo} from './meta-info';
import {Contact} from './contact';
import {OrderRegion} from './order-region';
import {ShopType} from './enum/shop-type.enum';

export class Shop extends BaseDto {

  owner: string;
  name: string;

  alias: string;
  longDesc: string;
  shortDesc: string;
  openingTime: string;
  privacy: string;
  actualInfo: string;
  minOrderValue = 0;

  extInfo: Array<ExtInfo> = [];
  metaInfo: MetaInfo;

  contact: Contact;

  orderRegions: Array<OrderRegion> = [];

  isOrderAble = true;
  needIndex = true;
  shopType = ShopType.RESTAURANT;

  latitude: string;
  longitude: string;
}
